lsusb.c

```diff
@@ -288,6 +288,32 @@ static void dump_device(
 		read_sysfs_prop(serial, sizeof(serial), sysfs_name, "serial");
 	}
 
+    int speed = libusb_get_device_speed(dev);
+    if (speed != LIBUSB_SPEED_UNKNOWN) {
+        const char *negotiated_speed = NULL;
+        switch (speed) {
+            case LIBUSB_SPEED_LOW     :
+                negotiated_speed = "Low Speed (1Mbps)";
+                break;
+            case LIBUSB_SPEED_FULL    :
+                negotiated_speed = "Full Speed (12Mbps)";
+                break;
+            case LIBUSB_SPEED_HIGH    :
+                negotiated_speed = "High Speed (480Mbps)";
+                break;
+            case LIBUSB_SPEED_SUPER   :
+                negotiated_speed = "SuperSpeed (5Gbps)";
+                break;
+            case LIBUSB_SPEED_SUPER_PLUS :
+                negotiated_speed = "SuperSpeed+ (10Gbps)";
+                break;
+            default:
+                negotiated_speed = "Unknown";
+                break;
+        }
+        printf("Negotiated speed :        %s\n", negotiated_speed);
```

Comment:
This will now be the second line of the output, are you sure that is ok?

Also, what happens if libusb can not figure out the speed, why not also print out "Unknown" as well instead of skipping it entirely?

Reply:
Question 1: It don't find it disturbing (but I am likely biased towards seeing the speed :P). I am happy to add the line wherever you deem fit the best.

Question: Done. Always showing it even in unknown.